client/src/utils/FetchUtil.ts

```diff
@@ -1,7 +1,34 @@
 import { DataFile, DataSource } from '@dekkai/data-source';
 import s3Client from '@/services/S3Service';
 
+const storeSizeLimit = 104857600; // 100MB in bytes
+const storeEntryLimit = 10485760; // 10MB in bytes
 const memoizedStore = new Map();
+const memoizedSize = new Map();
+
+const storeSize = (): number => {
+  let totalValue = 0;
+  for (const value of memoizedSize.values()) {
```

Comment:
Consider keeping track of size as items are added removed instead of recomputing each time.